proof/refine/ARM/KHeap_R.thy

```diff
@@ -3641,5 +3747,55 @@ lemma update_sc_no_reply_stack_update_corres:
    apply (wpsimp simp: obj_at_def is_sc_obj_def)+
   done
 
+lemma ko_at'_inj:
+  "ko_at' ko ptr  s \<Longrightarrow> ko_at' ko' ptr s \<Longrightarrow> ko' = ko"
+  by (clarsimp simp: obj_at'_real_def ko_wp_at'_def)
+
+(* FIXME RT: Move these to AInvs where possible *)
+definition injective_ref where
+  "injective_ref ref heap \<equiv> (\<forall>q p1 p2. (p1, ref) \<in> heap q \<and> (p2, ref) \<in> heap q \<longrightarrow> p1 = p2)"
```

Comment:
I think this `injective_ref` is more or less the same concept as `heap_refs_inj`  or similar related ones. There is `sym_refs_inj_sc_tcbs`, which might be quite close to `injective_ref_SCTcb` below. But maybe this one here is easy to use in any case. Could you create a JIRA ticket to investigate this and add that info to the FIXME?

Reply:
Yes, this acheives the same as the heap_refs version, but that version depends on heaps which are not defined for all references. It's definitely possible that I could replace `injective_ref_SCTcb` with `sym_refs_inj_sc_tcbs` . What kind of JIRA issue did you have in mind? Just doing that replacement?

Reply:
Doing some investigation and finding a most clean and minimal solution would be good.